### ISSUE TITLE
Lighting pitch ditches with European archers (AI)

### ISSUE THREAD

**Issue by Heroesflorian** [Sh0wdown/UnofficialCrusaderPatch#172]
Currently, the AI generally **cannot** light pitch ditches with European archers, and the AI generally **can** light pitch ditches with Arabian archers. 

This means the Abbot (who even has pitch ditches and braziers in his vanilla castles) can never ever use those, while a Snake or Sultan (who gather Arabian archers, if only for their attacks, on the keep) can use pitch ditches if their AIV has a brazier on the keep. This also means that even the Wolf cannot use his pitch ditches properly, only with Arabian archers from his keep or (by rare chance) on a gatehouse on their way out, if those happen to stand next to a brazier. 

Now, seeing that even some European AI characters with pitch ditches and braziers in their original castles cannot use their own pitch ditch, I would say this clearly is a bug missed by Firefly, and was not an intended "feature" or limitation. 
(And nope, neither Wolf nor Abbot have their castles optimized to use oil engineers for lighting pitch in any way either, unlike what we see for the Sheriff with his fire throwers.)

Soo... I think it would be really awesome to tell the AI that European archers with fire arrows can light pitch ditches just as well as Arabian archers! :)

**Comment by ByBurton** [comment]
@Sh0wdown This is a big issue that makes the pitch the Wolf uses useless in most cases. Can we have that fixed soon? Or if that may be too hard, maybe make him use arabian archers #165  on towers?

**Comment by Heroesflorian** [comment]
I do not see Arabian archers for the Wolf as a solution to the problem, though. 
That will only help the Wolf and simultaneously buff his "OP" defense even more, while leaving ~all other~ 7 of the other 9 European Lords out in the cold.

**Comment by MarcusVipsaniusAgrippa** [comment]
> 
> 
> I do not see Arabian archers for the Wolf as a solution to the problem, though.
> That will only help the Wolf and simultaneously buff his "OP" defense even more, while leaving all other 7 of the other 9 European Lords out in the cold.

I agree. I also don't think arabian archers would fit the wolf's castle.

**Comment by ByBurton** [comment]
> > I do not see Arabian archers for the Wolf as a solution to the problem, though.
> > That will only help the Wolf and simultaneously buff his "OP" defense even more, while leaving all other 7 of the other 9 European Lords out in the cold.
> 
> I agree. I also don't think arabian archers would fit the wolf's castle.

They would fit his castle. He already uses a lot of arabian archers.

**Comment by DarkThief-Darek** [comment]
@Sh0wdown I know you are currently busy with the AIC implementation into the UCP, which will be an awesome addition! Thanks for your and ByBurtons work!

With the addition of the AIC Editor and my small project idea I need European Archers to light pitch ditches, because I don't want to give all my reworked AIs Arabian Archers for that.

It would be awesome if you can fix this in the near furture, so we have more flexibility with the upcoming AIC tool.

**Comment by ByBurton** [comment]
@Sh0wdown is this difficult to change? I see you could edit fix the fire ballistae not shooting at monks, so can you also fix this?

**Comment by Krarilotus** [comment]
I believe the issue here is, that arabian archers have the agressive stance by default while european archers are on position hold stance. Only archers in aggressive stance will light the pitch automatically, when a certain combined unit hitpoint threshhold is moving towards that pitchtrap

**Comment by ByBurton** [comment]
@Krarilotus Really? I never looked at the stances of defensive units and assumed they all were aggressive ...
And do human-controlled archers also do that if you set them on aggressive?!

**Comment by Heroesflorian** [comment]
> I believe the issue here is, that arabian archers have the agressive stance by default while european archers are on position hold stance. Only archers in aggressive stance will light the pitch automatically, when a certain combined unit hitpoint threshhold is moving towards that pitchtrap

Interesting... although I've noticed Saladin lights up pitch even for individual slaves sometimes... is there any more info known regarding that "combined hitpoint threshold" (does it differ between AI characters, how difficult would it be to change it, is it affected by the size of the pitch trap in question, ...)?

**Comment by J-T-de** [comment]
Krarilotus and I tested it some time age, it was not the aggressive stance thing.

**Comment by LordHansCapon** [comment]
@Heroesflorian A fix for this bug is in testing phase. The problem was that the Crusader Archers are forced to NOT light pitches. The check function for pitch lighting was missing basically, that has been placed into the archer's AI just like the Arabian archer's.
